src/arrayMethodSort.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction) {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - i - 1; j++) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            const temporary = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temporary;
+          }
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length - 1; i++) {
+        for (let j = 0; j < this.length - i - 1; j++) {
+          if (String(this[j]) > String(this[j + 1])) {
+            const temporary = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temporary;
+          }
+        }
+      }
+    }
```

Comment:
let's improve your solution )
 - DRY - don't repeat yourself 
 - you have a lot of code duplications
 - for fixing that you can set for the compareFunction a default value, for example 
 
<img width="1030" alt="Screenshot 2023-07-09 at 16 28 49" src="https://github.com/mate-academy/js_array-method-sort/assets/64746943/f8e6cd3b-6580-42ad-8857-620786dd2b90">